lsp-types/src/Language/LSP/VFS.hs

```diff
@@ -357,6 +348,7 @@ getCompletionPrefix pos@(J.Position l c) (VirtualFile _ _ ropetext) =
             let modParts = dropWhile (not . isUpper . T.head)
                                 $ reverse $ filter (not .T.null) xs
                 modName = T.intercalate "." modParts
+            curLine <- headMaybe $ T.lines $ Rope.toText curRope
```

Comment:
Why not use `Rope.lines`? Won't that be faster?

Reply:
`curRope` is already a single line, but it likely includes an enclosing `\n`. The only purpose of `headMaybe . T.lines . Rope.toText` is to strip this `\n`. I've changed it to a more explicit version.